src/goto-programs/goto_convert.cpp

```diff
@@ -344,14 +344,40 @@ void goto_convertt::convert_label(
 
   goto_programt tmp;
 
-  // magic thread creation label?
+  // magic thread creation label.
+  // The "__CPROVER_ASYNC_" signals the start
```

Comment:
It's actually the single statement that's labelled in this way that is the new thread. Typically, this statement will be a function call.

Reply:
can you clarify this comment please? There is nothing preventing you from having something like __CPROVER_ASYNC: statement1, statement2, statement3, ... ;

Reply:
If you use a comma, then those are expressions, not statements.